internal/legacy/legacy.go

```diff
@@ -89,19 +92,25 @@ func (c *CLIWrapper) cacheDir() string {
 	return path.Join(os.TempDir(), fmt.Sprintf("%s-%s-%s", c.Config.Application.Slug, PHPVersion, LegacyCLIVersion))
 }
 
-// Init the CLI wrapper, creating a temporary directory and copying over files
-func (c *CLIWrapper) Init() error {
+// init initializes the CLI wrapper, creating a temporary directory and copying over files.
+func (c *CLIWrapper) init() error {
+	if c.initialized {
+		return nil
+	}
```

Comment:
It would be best to use something like `sync.OnceFunc`[^1] here  to guard against race conditions and make sure the initialization is only run once.

[^1]: https://pkg.go.dev/sync#OnceFunc